Docs/Book/Cookbook_v2.rst

```diff
@@ -0,0 +1,1009 @@
+
+.. testcode::
+
+   from rdkit import Chem
+   # default RDKit behavior is to reject hypervalent P, so you need to set sanitize=False
+   m = Chem.MolFromSmiles('F[P-](F)(F)(F)(F)F.CN(C)C(F)=[N+](C)C',sanitize=False)
+
+.. testcode::
+
+   # next, you probably want to at least do a partial sanitization so that the molecule is actually useful:
+   m.UpdatePropertyCache(strict=False)
+   Chem.SanitizeMol(m,Chem.SanitizeFlags.SANITIZE_FINDRADICALS|Chem.SanitizeFlags.SANITIZE_KEKULIZE|Chem.SanitizeFlags.SANITIZE_SETAROMATICITY|Chem.SanitizeFlags.SANITIZE_SETCONJUGATION|Chem.SanitizeFlags.SANITIZE_SETHYBRIDIZATION|Chem.SanitizeFlags.SANITIZE_SYMMRINGS,catchErrors=True)
+
+
+Capturing Error Messages with Chem.DetectChemistryProblems
+==========================================================
+
+| **Author:** Greg Landrum
+| **Source:** `<https://sourceforge.net/p/rdkit/mailman/message/36779572/>`_
+| **Index ID#:** RDKitCB_14
+| **Summary:** Identify and capture error messages when creating mol objects.
+
+.. testcode::
+
+   from rdkit import Chem
+   m = Chem.MolFromSmiles('CN(C)(C)C',sanitize=False)
+   problems = Chem.DetectChemistryProblems(m)
+   print(len(problems))
+
+.. testoutput::
+
+   1
+
+.. testcode::
+
+   print(problems[0].GetType())
+   print(problems[0].GetAtomIdx())
+   print(problems[0].Message())
+
+.. testoutput::
+
+   AtomValenceException
+   1
+   Explicit valence for atom # 1 N, 4, is greater than permitted
+
+.. testcode:: 
+
+   m2 = Chem.MolFromSmiles('c1cncc1',sanitize=False)
+   problems = Chem.DetectChemistryProblems(m2)
+   print(len(problems))
+
+.. testoutput::
+
+   1
+
+.. testcode::
+
+   print(problems[0].GetType())
+   print(problems[0].GetAtomIndices())
+   print(problems[0].Message())
+   
+.. testoutput::
+   :options: -ELLIPSIS, +NORMALIZE_WHITESPACE
+
+   KekulizeException
+   (0, 1, 2, 3, 4)
+   Can't kekulize mol.  Unkekulized atoms: 0 1 2 3 4
+
+Miscellaneous Topics
+**********************
+
+Explicit Valence and Number of Hydrogens
+==============================================
+
+| **Author:** Michael Palmer/ Greg Landrum
+| **Source:** `<https://sourceforge.net/p/rdkit/mailman/message/29679834/>`_
+| **Index ID#:** RDKitCB_11
+| **Summary:** Calculate the explicit valence, number of explicit and implicit hydrogens, and total number of hydrogens on an atom. See the link for an important explanation about terminology and implementation of these methods in RDKit. Highlights are presented below.
+
+Most of the time (exception is explained below), explicit refers to atoms that are in the graph and 
+implicit refers to atoms that are not in the graph (i.e., Hydrogens). So given that the ring is aromatic (e.g.,in pyrrole), 
+the explicit valence of each of the atoms (ignores the Hs that are not present in the graph) in pyrrole is 3. If you want the Hydrogen count,
```

Comment:
This reads more smoothly to me if you change "ignores" to "ignoring".

Reply:
Thanks, changed it.